### PR TITLE
DOCS-886: Improve Go modular resource code template by adding module with contextual main

### PR DESCRIPTION
**Pull request by sguequierre**
* Apply Alan's feedback for compiling a go module 
* Add code in from the Viam examples repo's [complexmodule module.go](https://github.com/viamrobotics/rdk/blob/a0dec7d74085139fe77a637335ba43640355ca47/examples/customresources/demos/complexmodule/module.go) 
* Try to clear up module execution

### CODE REVIEW

File: docs/extend/modular-resources/_index.md [hunk 1]
```diff
@@ -110,9 +110,12 @@ For example:
 {{< tabs name="Base Model Modules" >}}
 {{% tab name="Go" %}}
 
-This example module code is adapted from the full demo module available on the [Viam GitHub](https://github.com/viamrobotics/rdk/blob/main/examples/customresources/models/mybase/mybase.go), and creates a singular modular resource implementing Viam's built-in Base API (rdk:service:base).
+This example custom model code (<file>mybase.go</file>) and module (<file>mymodule.go</file>) is adapted from the full demo module available on the [Viam GitHub](https://github.com/viamrobotics/rdk/blob/main/examples/customresources/models/mybase/mybase.go), and creates a singular modular resource implementing Viam's built-in Base API (rdk:service:base).
```
Comment (andf-viam) [inline docs/extend/modular-resources/_index.md:113]:
```suggestion
The example custom model code (<file>mybase.go</file>) and module (<file>mymodule.go</file>) are adapted from the full demo module available on the [Viam GitHub repository](https://github.com/viamrobotics/rdk/blob/main/examples/customresources/models/mybase/mybase.go), and creates a singular modular resource implementing Viam's built-in Base API (rdk:service:base).
```

supernit: The above aligns plural (`the ... code ... and module are adapted`) and article (`the ... repo`). The article change also applicable to the Python tab, if you like!

Comment (penguinland) [inline docs/extend/modular-resources/_index.md]: I recognize that my comment is about a part of the line you didn't edit, but what does "singular" mean in this context? It's not the only one, it's not the best one, it's not a unique one. Could we remove that word and just say it "creates a modular resource"?

**Review by andf-viam — APPROVED** [review]
LGTM, one tiny nit! Thanks!

File: docs/extend/modular-resources/_index.md [hunk 2]
```diff
@@ -279,6 +282,55 @@ func init() {
...
+
+    // NOTE: You must update the following line to import your local package "mybase"
+    "go.viam.com/rdk/examples/customresources/models/mybase"
+)
+
+func main() {
+    // NewLoggerFromArgs will create a golog.Logger at "DebugLevel" if
+    // "--log-level=debug" is the third argument in os.Args and at "InfoLevel"
+    // otherwise.
+    utils.ContextualMain(mainWithArgs, module.NewLoggerFromArgs("complexmodule"))
```
Comment (sguequierre) [inline docs/extend/modular-resources/_index.md:308]:
```suggestion
    utils.ContextualMain(mainWithArgs, module.NewLoggerFromArgs("yourmodule"))
```

File: docs/extend/modular-resources/_index.md [hunk 3]
```diff
@@ -279,6 +282,55 @@ func init() {
...
+import (
+    "context"
+
+    "github.com/edaniels/golog"
+
+    "go.viam.com/rdk/components/base"
+    "go.viam.com/rdk/module"
+    "go.viam.com/utils"
+
+    // NOTE: You must update the following line to import your local package "mybase"
```
Comment (penguinland) [inline docs/extend/modular-resources/_index.md:300]: :+1: Excellent comment here!

Comment (sguequierre) [inline docs/extend/modular-resources/_index.md:300]: @penguinland LMK if this does not provide enough info here-- would this be the path to mybase after compiling?

File: docs/extend/modular-resources/_index.md [hunk 4]
```diff
@@ -449,6 +501,7 @@ To add a module to your robot, you need to have an [executable file](https://en.
 Your options for completing this step are flexible, as this file does not need to be in a raw binary format.
 
 If using the Go SDK, Go will build a binary when you compile your module.
+Expand the [Go module examples](#code-your-module) to view <file>mymodule.go</file> as an example of a module you can run that initializes a custom model <file>myBase.go</file> from the registry.
```
Comment (sguequierre) [inline docs/extend/modular-resources/_index.md:504]:
```suggestion
Expand the [Go module examples](#code-your-module) to view <file>mymodule.go</file> as an example of a module you can run that initializes a custom model <file>mybase.go</file> from the registry.
```

File: docs/extend/modular-resources/_index.md [hunk 5]
```diff
@@ -279,6 +282,55 @@ func init() {
...
+    "go.viam.com/rdk/module"
+    "go.viam.com/utils"
+
+    // NOTE: You must update the following line to import your local package "mybase"
+    "go.viam.com/rdk/examples/customresources/models/mybase"
+)
+
+func main() {
+    // NewLoggerFromArgs will create a golog.Logger at "DebugLevel" if
+    // "--log-level=debug" is the third argument in os.Args and at "InfoLevel"
```
Comment (penguinland) [inline docs/extend/modular-resources/_index.md:306]: Not sure about this "third argument" part. Shouldn't the argument flags work in any order? Hopefully it'll create the logger at the right debug level if that arg is present anywhere in `os.Args`.

**Review by penguinland — APPROVED** [review]
All of my feedback is optional nits. LGTM!

File: docs/extend/modular-resources/_index.md (hunk 3 above)
Comment (penguinland) [reply docs/extend/modular-resources/_index.md:300]: "After compiling" doesn't really make sense. It's the path to mybase, full stop. I like the comment: it's saying "you probably changed the package name when you made your own version of the code, so remember to change this line to match."

File: docs/extend/modular-resources/_index.md (hunk 5 above)
Comment (sguequierre) [reply docs/extend/modular-resources/_index.md:306]: @penguinland  I adapted this sample from the RDK demos and this note came w/ --> I agree it seems misleading if the flag is what's triggering 
Making this change here

File: docs/extend/modular-resources/_index.md [hunk 6]
```diff
@@ -110,9 +110,12 @@ For example:
 {{< tabs name="Base Model Modules" >}}
 {{% tab name="Go" %}}
 
-This example module code is adapted from the full demo module available on the [Viam GitHub](https://github.com/viamrobotics/rdk/blob/main/examples/customresources/models/mybase/mybase.go), and creates a singular modular resource implementing Viam's built-in Base API (rdk:service:base).
+The example custom model code (<file>mybase.go</file>) and module (<file>mymodule.go</file>) are adapted from the full demo module available on the [Viam GitHub](https://github.com/viamrobotics/rdk/blob/main/examples/customresources/models/mybase/mybase.go), and creates a modular resource implementing Viam's built-in Base API (rdk:service:base).
```
Comment (sguequierre) [inline docs/extend/modular-resources/_index.md:113]:
```suggestion
The code for this custom model (<file>mybase.go</file>) and module (<file>mymodule.go</file>) is adapted from the full demo modules available on the [Viam GitHub](https://github.com/viamrobotics/rdk/blob/main/examples/customresources/models/mybase/mybase.go).
The module registers a modular resource implementing Viam's built-in Base API (rdk:service:base).
```

**Review by penguinland — APPROVED** [review]
You made changes since I reviewed, but everything still LGTM. If you're waiting on me to approve again, you've got it.

**Pull request merged**